coverity-scan.sh

```diff
@@ -1,62 +1,167 @@
 #!/usr/bin/env bash
 # Coverity scan script
 #
-# To run this script you need to provide API token. This can be done either by:
-#  - Putting token in ".coverity-token" file
-#  - Assigning token value to COVERITY_SCAN_TOKEN environment variable
-#
 # Copyright: SPDX-License-Identifier: GPL-3.0-or-later
 #
 # Author  : Costa Tsaousis (costa@netdata.cloud)
 # Author  : Pawel Krupa (paulfantom)
 # Author  : Pavlos Emm. Katsoulakis (paul@netdata.cloud)
 
-cpus=$(grep -c ^processor </proc/cpuinfo)
+# To run manually, save configuration to .coverity-scan.conf like this:
+#
+# the repository to report to coverity - devs can set here their own fork
+# REPOSITORY="netdata/netdata"
+#
+# the email of the developer, as given to coverity
+# COVERITY_SCAN_SUBMIT_MAIL="you@example.com"
+#
+# the token given by coverity to the developer
+# COVERITY_SCAN_TOKEN="TOKEN taken from Coverity site"
+#
+# the absolute path of the cov-build - optional
+# COVERITY_BUILD_PATH="/opt/cov-analysis-linux64-2019.03/bin/cov-build"
+#
+# when set, the script will print on screen the curl command that submits the build to coverity
+# this includes the token, so the default is not to print it.
+# COVERITY_SUBMIT_DEBUG=1
+#
+# All these variables can also be exported before running this script.
+#
+# If the first parameter of this script is "install",
+# coverity build tools will be downloaded and installed in /opt/coverity
+
+# the version of coverity to use
+COVERITY_BUILD_VERSION="cov-analysis-linux64-2019.03"
+
+source packaging/installer/functions.sh || exit 1
+
+cpus=$(find_processors)
 [ -z "${cpus}" ] && cpus=1
 
+if [ -f ".coverity-scan.conf" ]
+then
+	source ".coverity-scan.conf" || exit 1
+fi
```

Comment:
I would probably write it as: `[ -f ".coverity-scan.conf" ] && source ".coverity-scan.conf"`, but that may be just matter of personal preference.

Also instead of the `exit 1` logic, is better to just set `set -e` to enable strict error checking and just use the or option for command we don't want them to stop the process if they fail. what do you think?